README.md

```diff
@@ -169,14 +214,20 @@ The contents of the User-Agent header of the request.
 
 ### morgan.compile(format)
 
-Compile a format string into a function for use by `morgan`. A format string
+Compile a format string into a format function for use by `morgan`. A format string
 is a string that represents a single log line and can utilize token syntax.
 Tokens are references by `:token-name`. If tokens accept arguments, they can
 be passed using `[]`, for example: `:token-name[pretty]` would pass the string
 `'pretty'` as an argument to the token `token-name`.
 
+The function returned from `morgan.compile` takes three arguments `tokens` , `req`, and `res`
+where `tokens` represents an object with all known tokens. If a log should be skipped the
+function will return `null`.
+
 Normally formats are defined using `morgan.format(name, format)`, but for certain
-advanced uses, this compile function is directly available.
+advanced uses, this compile function is directly available. The function returned can be passed directly to morgan using `morgan(morgan.compile(format))`. In other words, `morgan.compile` is a quick short-hand to turn format strings into format functions.
+
+In some rare cases (such as [outputting JSON logs from `morgan`](#output-json-logs)) you may want to write your own custom format function. These functions should have the same API as functions returned from `morgan.compile` accepting three arguments: `tokens`, `req`, and `res`.
```

Comment:
I think this sentence is probably unnecessary with the added information now found at the top of the readme.